### PR TITLE
Add limit_column_characters function with tests

### PR DESCRIPTION
**Pull request by szuckerman**
As per [issue 15](https://github.com/ericmjl/pyjanitor/issues/15) I've added a limit_column_characters function.

There's three tests; they should be sufficient.

I don't think we could have gone with the suggestion to use the df_column_uniquify function in [this stackoverflow answer](https://stackoverflow.com/a/44957247) since it will only work if the duplicate columns are next to each other. If they skip and repeat later, the count starts again.

### CODE REVIEW

File: janitor/functions.py [hunk 1]
```diff
@@ -788,3 +788,55 @@ def add_column(df, colname: str, value):
...
+      first distinct column will be 10 characters and the remaining will be 12
+      characters (assuming a column separator of one character)
+    :param col_separator: The separator to use for counting distinct column values.
+      I think an underscore looks nicest, however a period is a common option as well.
+      Supply an empty string (i.e. '') to remove the separator.
+
+    """
+
+    assert isinstance(column_length, int), "`column_length` must be an integer!"
+    assert isinstance(col_separator, str), "`col_separator` must be a string!"
```
Comment (ericmjl) [inline janitor/functions.py]: Very nice! I'm always a fan of defensive programming checks :smile:

**Comment by szuckerman** [comment]
Sure! I'll run it through `black` and update the PR. Might not get to it today though; `black` is on my list for new tools to try out 😃 .

**Comment by szuckerman** [comment]
Looks like `black` has an 88 character limit; gotta change it to 79 for Travis.

**Comment by ericmjl** [comment]
Right: `black -l 79 .` should do the trick.

**Comment by szuckerman** [comment]
I'm really not sure why the Travis build keeps failing... Are the previous commits making the problem?

**Comment by ericmjl** [comment]
@szuckerman this looks like an error on my side. Your PR looks good to me at this point. I just have to fix something underlying, I guess.

**Comment by ericmjl** [comment]
As things turn out, it was an issue with conda. Let me fix this on the PR.

**Comment by ericmjl** [comment]
Woohoo, everything looks good to me! Merging in. Thanks @szuckerman for your contribution!

**Pull request merged**